torch/_inductor/lowering.py

```diff
@@ -2163,8 +2163,13 @@ def fn(idx):
         assert len(idx) == len(output_size)
         assert len(indices_loaders) == len(indexed_size)
         new_index = [
-            ops.indirect_indexing(
-                loader(idx[start_offset:end_offset]), size, check=check
+            ir.ModularIndexing(
```

Comment:
as @peterbell10 pointed out, this approach will unconditionally put even invalid indices in the valid range, you can't do that.

Reply:
> as @peterbell10 pointed out, this approach will unconditionally put even invalid indices in the valid range, you can't do that.

Yes, I guess we also need to improve `index_propagation.py` to simplify the added checks when the indices are constants...